pkg/controller/proxyconfig/validation.go

```diff
@@ -159,21 +177,41 @@ func (r *ReconcileProxyConfig) validateSystemTrustBundle(trustBundle string) ([]
 	return bundleData, nil
 }
 
-// validateHTTPReadinessEndpoint validates an http readinessEndpoint endpoint.
-func validateHTTPReadinessEndpoint(httpProxy, endpoint string) error {
-	if err := validateHTTPReadinessEndpointWithRetries(httpProxy, endpoint, proxyProbeMaxRetries); err != nil {
+// validateReadinessEndpoint validates endpoint using proxy. If caBundle
+// is not nil, TLS is used for the probe with caBundle as the cert pool.
+func validateReadinessEndpoint(caBundle []*x509.Certificate, proxy, endpoint string) error {
+	proxyURL, err := url.Parse(proxy)
+	if err != nil {
+		return fmt.Errorf("failed to parse proxy url '%s': %v", proxy, err)
+	}
+
+	endpointURL, err := url.Parse(endpoint)
+	if err != nil {
+		return fmt.Errorf("failed to parse endpoint url '%s': %v", endpoint, err)
+	}
+
+	if (endpointURL.Scheme == proxyHTTPScheme && proxyURL.Scheme == proxyHTTPSScheme) ||
+		(endpointURL.Scheme == proxyHTTPSScheme && proxyURL.Scheme == proxyHTTPScheme) {
```

Comment:
was one of these pairings supposed to be http?

Reply:
I tried to combined the 2 checks into 1. I'm going to separate.